runelite-client/src/main/java/net/runelite/client/plugins/bankvalue/BankValuePlugin.java

```diff
@@ -0,0 +1,381 @@
+
+	@Provides
+	BankValueConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(BankValueConfig.class);
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (!event.getGroup().equals("Bank Value"))
+		{
+			return;
+		}
+
+		// When changing the configuration, just reset the call.
+		// Items should be cached so it should be relatively quick to recalculate.
+		// This is mostly to avoid wrong additions while looping and getting the prices while changing
+		// the config multiple times.
+		if ((event.getKey().equals("showGE") || event.getKey().equals("showHA")) && bankActive)
+		{
+			stopThreads();
+			resetBankTitle();
+			calculateBankWithThread();
+		}
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		stopThreads();
+		resetBankTitle();
+	}
+
+	@Override
+	protected void startUp()
+	{
+		Widget widgetBankTitleBar = client.getWidget(WidgetInfo.BANK_TITLE_BAR);
+
+		if (widgetBankTitleBar == null || widgetBankTitleBar.isHidden())
+		{
+			return;
+		}
+
+		calculateBankWithThread();
+	}
+
+	@Subscribe
+	public void onMenuChanged(MenuOptionClicked event)
+	{
+		if (!bankActive)
+		{
+			return;
+		}
+
+		String strMenuOption = event.getMenuOption();
+
+		if (strMenuOption.contains("Withdraw") ||
+				strMenuOption.contains("Deposit") ||
+				strMenuOption.contains("View tab") ||
+				strMenuOption.contains("View all"))
+		{
+			stopThreads();
+			calculateBankWithThread();
+		}
+	}
+
+	@Subscribe
+	public void onWidgetHidden(WidgetHiddenChanged event)
+	{
+		// Below is several checks to hopefully get a bank opening/closing event.
+		// If there are better events to check for this, then this should be changed.
+		// Then we we can remove the sleep() in calculateBankThread()
+		Widget widget = event.getWidget();
+
+		int group = WidgetInfo.TO_GROUP(widget.getId());
+
+		// This seems to be the only event sent when closing the bank
+		if (bankActive && group == WidgetID.RESIZABLE_VIEWPORT_BOTTOM_LINE_GROUP_ID && event.isHidden())
+		{
+			log.debug("Bank closing?");
+			stopThreads();
+			return;
+		}
+
+		// This event gets called a lot, so check to also see if it's are already bankActive
+		// This still does not work great, because it still gets called twice, but the first time is just a dud.
+		if (!bankActive && group == WidgetID.BANK_GROUP_ID)
+		{
+			calculateBankWithThread();
+		}
+	}
+
+	private void stopThreads()
+	{
+		bankActive = false;
+		log.debug("Bank value stopping");
+
+		if (executor != null)
+		{
+			executor.shutdownNow();
+		}
+
+		if (bankThread != null)
+		{
+			bankThread.interrupt();
+		}
+	}
+
+	private void calculateBankWithThread()
+	{
+		bankActive = true;
+		log.debug("Bank value running");
+		bankThread = new Thread(this::calculateBank);
+		bankThread.start();
+	}
+
+	class ItemPriceAndAlch extends ItemPrice
+	{
+		@Getter
+		@Setter
+		private int highAlchPrice;
+
+		ItemPriceAndAlch(int gePrice, int highAlchPrice)
+		{
+			setPrice(gePrice);
```

Comment:
Instead of doing this, add @Value annotation on ItemPriceAndAlch, then there would be no need for adding getter and setter annotation and making constructor